rcl/src/rcl/client.c

```diff
@@ -111,6 +112,14 @@ rcl_client_init(
     RCL_SET_ERROR_MSG(rmw_get_error_string().str);
     goto fail;
   }
+  // get actual qos, and store it
+  rmw_ret_t rmw_ret = rmw_client_get_actual_qos(
```

Comment:
if request data writer and response data reader actual qos are different, what we should do? there is only one qos setting for client, so i am expecting they must match? if so, we can just return the error code to fail the client initialization? i am not sure about this, i would like to hear the opinion.